### DISCUSSION TITLE
Localized routing not working for dynamic routes

### DISCUSSION THREAD

**Discussion by plushdohn** [vercel/next.js#47360] [Help]
### Summary

I have a Next app which only has two routes: `index.tsx` and `[slug].tsx`. The slug route uses `getStaticProps` and `getStaticPaths` to fetch the page from a CMS.

I have enabled internationalized routing by adding this into my `next.config.js`:
```js
i18n: {
  locales: ["en", "fr"],
  defaultLocale: "en",
}
```

The `index` route works fine, if I go to `/` with an `Accept-Language` header of `fr-FR` I get redirected to the `/fr` subroute. The dynamic route `[slug].tsx` doesn't do the same. If I go straight to it, it doesn't redirect me to the `fr` locale version. And printing the request `locale` in `getStaticProps` shows the default `en` locale, instead of the one computed from my `Accept-Language` header.

I'm aware people usually forget to generate the locale paths in `getStaticPaths` and I don't think that's the problem, here's how I'm generating them:
```js
const paths = locales
  .map((locale) => posts.map((p) => ({ params: { slug: p.id }, locale })))
  .flat();

// { params: { slug: "post1" }, locale: "en" }
// { params: { slug: "post1" }, locale: "fr" }
```

What am I missing? Thanks!

**Comment by kboskin** [comment]
Facing the same. Setup


    "next": "^14.2.4",
    "next-i18next": "^15.3.0",
